chia/server/server.py

```diff
@@ -258,7 +260,10 @@ async def garbage_collect_connections_task(self) -> None:
                             to_remove.append(connection)
             for connection in to_remove:
                 self.log.debug(f"Garbage collecting connection {connection.peer_host} due to inactivity")
-                await connection.close()
+                if connection.closed:
+                    self.all_connections.pop(connection.peer_node_id)
+                else:
+                    await connection.close()
```

Comment:
I'm guessing there's a reason we shouldn't pop it when we close it?

```suggestion
                if not connection.closed:
                    await connection.close()
                self.all_connections.pop(connection.peer_node_id)
```

Reply:
They get popped when the close() call comes back around and calls the close_callback - so I didn't change how this part of the code worked.